test/verify/check-ws-bastion

```diff
@@ -0,0 +1,133 @@
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+import parent
+from testlib import *
+from testvm import DEFAULT_IMAGE
+
+HOST = "host.containers.internal"
+
+
+@unittest.skipUnless(DEFAULT_IMAGE == "fedora-coreos", "no cockpit/ws container on this image")
+@nondestructive
+class TestWsBastionContainer(MachineCase):
+    def setUp(self):
+        super().setUp()
+        # stop ws container from previous runs
+        self.machine.stop_cockpit()
+        # undo cockpit/ws install steps
+        self.restore_file("/etc/systemd/system/cockpit.service")
+        self.machine.execute("rm /etc/systemd/system/cockpit.service")
+        self.addCleanup(self.machine.execute, "podman rm -f --all")
+
+    def approve_key(self, b, hostname):
+        b.wait_visible("#hostkey-group")
+        b.wait_in_text("#hostkey-message-1", f"You are connecting to {hostname} for the first time.")
+        b.click("#login-button")
+
+    def testPasswordLogin(self):
+        m = self.machine
+        b = self.browser
+        m.execute("podman run -d --name cockpit-bastion -p 9090:9090 localhost/cockpit/ws")
+        m.execute("until curl --fail --head -k https://localhost:9090/; do sleep 1; done")
+
+        b.ignore_ssl_certificate_errors(True)
+        b.open("/", tls=True)
+
+        b.wait_visible("#login")
+        # should be pre-configured to RequireHost
+        b.wait_not_visible("#option-group")
+        b.wait_visible("#server-field")
+        # LoginTitle from default-bastion.conf
+        b.wait_text("#server-name", "Cockpit Bastion")
+        # No branding by default
+        b.wait_text("#brand", "")
+
+        b.set_val("#login-user-input", "admin")
+        b.set_val("#login-password-input", "foobar")
+
+        # Requires a host
+        b.click("#login-button")
+        b.wait_in_text("#login-error-message", "host to connect")
+        # so connect to our own container host
+        b.set_val("#server-field", HOST)
+        b.set_val("#login-password-input", "foobar")
+        # key is unknown
+        b.click("#login-button")
+        self.approve_key(b, HOST)
+
+        b.wait_visible('#content')
+        b.wait_text('#current-username', 'admin')
+        b.logout()
+
+        # remembers the last host via URL, server field should be pre-filled
+        self.assertEqual(b.eval_js("window.location.pathname"), f"/={HOST}/system")
+        # FIXME: login page does not really set this in the DOM? DOM has empty value, but browser shows the value
```

Comment:
What? It is not in dom, but it is visible? How does that work?

Reply:
I have no idea.. I stared at this for 30 mins, then gave up (as our bastion test doesn't cover this either), and thought let's not block on that..